src/util/extent.js

```diff
@@ -1,57 +1,105 @@
-        data = [data];
-    }
-    if (arguments.length === 2) {
-        // the fields parameter must be an array of field names, but we can pass non-array types in
-        if (!Array.isArray(fields)) {
-            fields = [fields];
+export default function() {
+
+    var include = noop,
+        pad = d3.functor(0);
+
+    /**
+    * @param {array} data an array of data points, or an array of arrays of data points
+    * @param {array} fields the names of object properties that represent field values, or accessor functions.
+    */
+    var extents = function(data, fields) {
+
+        // we need an array of arrays if we don't have one already
+        if (!Array.isArray(data[0])) {
+            data = [data];
         }
-    } else {
-        // for > 2 args, construct the fields
-        var args = Array.prototype.slice.call(arguments);
-        fields = args.slice(1);
-    }
-
-    // the fields can be a mixed array of property names or accessor functions
-    fields = fields.map(function(field) {
-        if (typeof field !== 'string') {
-            return field;
+        if (arguments.length === 2) {
+            // the fields parameter must be an array of field names,
+            // but we can pass non-array types in
+            if (!Array.isArray(fields)) {
+                fields = [fields];
+            }
+        } else {
+            // for > 2 args, construct the fields
+            var args = Array.prototype.slice.call(arguments);
+            fields = args.slice(1);
         }
-        return function(d) {
-            return d[field];
-        };
-    });
-
-    // Return the smallest and largest
-    return [
-        d3.min(data, function(d0) {
+
+        // the fields can be a mixed array of property names or accessor functions
+        fields = fields.map(function(field) {
+            if (typeof field !== 'string') {
+                return field;
+            }
+            return function(d) {
+                return d[field];
+            };
+        });
+
+        var dataMin = d3.min(data, function(d0) {
             return d3.min(d0, function(d1) {
                 return d3.min(fields.map(function(f) {
                     return f(d1);
                 }));
             });
-        }),
-        d3.max(data, function(d0) {
+        });
+
+        var dataMax = d3.max(data, function(d0) {
             return d3.max(d0, function(d1) {
                 return d3.max(fields.map(function(f) {
                     return f(d1);
                 }));
             });
-        })
-    ];
+        });
+
+        var min = dataMin;
+        var max = dataMax;
+
+        // Scale the range for the given padding
+        if (typeof dataMin === 'number' && typeof dataMax === 'number') {
+            var rangeDelta = pad() * (dataMax - dataMin) / 2;
+
+            min -= rangeDelta;
+            max += rangeDelta;
+        }
+
+        // Include the specified point in the range
+        var extraPoint = include();
```

Comment:
using the fact that `>` and `<` comparisons with `undefined` are always `false` feels a bit cryptic to me!